cmds/graph.c

```diff
@@ -871,21 +876,28 @@ static bool print_task_node(struct uftrace_task *task,
 static int graph_print_task(struct uftrace_data *handle, struct opts *opts)
 {
 	bool *indent_mask;
+	struct uftrace_task *task;
 
 	if (uftrace_done)
 		return 0;
 
 	if (handle->nr_tasks <= 0)
 		return 0;
 
+	task = handle->sessions.first_task;
+
 	setup_field(&output_task_fields, opts, &setup_task_field,
 		    field_task_table, ARRAY_SIZE(field_task_table));
 
 	pr_out("========== TASK GRAPH ==========\n");
 	print_header(&output_task_fields, "# ", "TASK NAME", 2);
 
 	indent_mask = xcalloc(handle->nr_tasks, sizeof(*indent_mask));
-	print_task_node(handle->sessions.first_task, NULL, indent_mask, 0);
+
+	/* filter out if total time is less than time-filter */
+	if (opts->threshold <= task->time.run)
+		print_task_node(task, NULL, indent_mask, 0, opts);
+
```

Comment:
Not sure it'll work well for complex data.  I'm concerning the lines..  Did you test it?

Reply:
Yes, I tested this with chrome data and other data as well.